modules/gdscript/doc_classes/GDScript.xml

```diff
@@ -23,5 +23,11 @@
 				[/codeblock]
 			</description>
 		</method>
+		<method name="unload_static" qualifiers="const">
+			<return type="void" />
+			<description>
+				Releases the persistent reference for the script if it has static variables. If no other references are kept, the script will be freed and modifications to static variables will be lost.
```

Comment:
To properly answer: this function removes the persistent reference from the cache and you cannot add it back. However, if the script is freed because of lack of references and loaded again, the persistent reference will be restored.

If you use the annotation, the persistent reference will never be created.

I particularly don't see why you would need to undo this action. If the script can be freed then it should not be a problem if it actually is. In which situation would one decide to keep it after releasing it? There's always the workaround of keeping a reference manually in a singleton or something.

We can unexpose if it's not needed at all. But I can't see a need to restore this after being released.

Reply:
> However, if the script is freed because of lack of references and loaded again, the persistent reference will be restored.

This should be reflected in the docs.

> We can unexpose if it's not needed at all.

I would prefer not to add this method yet. We can do this later if needed. The annotation has a clear use: you either initially worry about the static data of the script, or not, and do not change your mind at runtime. The method does not allow you to instantly clear data, predict when it will happen, or get the current status of the persistent reference. So I believe that it is unlikely to be useful to users, but can be confusing.